pkg/modprovider/child.go

```diff
@@ -217,25 +217,89 @@ func (h *childHandler) Diff(
 		resp.Changes = pulumirpc.DiffResponse_DIFF_NONE
 	case tfsandbox.Update:
 		resp.Changes = pulumirpc.DiffResponse_DIFF_SOME
+
 		// TODO[pulumi/pulumi-terraform-module#100] populate resp.Diffs
+		//
+		// Populating resp.Diffs does not seem to hurt, it will be made obsolete though by implementing
+		// DetailedDiff.
+		changed, err := h.computeChangedKeys(req)
+		if err != nil {
+			return nil, err
+		}
+		resp.Diffs = propertyKeysAsStrings(changed)
+
 	case tfsandbox.Replace, tfsandbox.ReplaceDestroyBeforeCreate:
 		resp.Changes = pulumirpc.DiffResponse_DIFF_SOME
+
+		// As discussed in https://github.com/pulumi/pulumi/issues/19103 getting this right does not yet affect
+		// the visual diff during pulumi preview, it only affects the ordering of Delete, Create calls which is
+		// immaterial here, and the sequencing of "deleting original", "creating replacement" messages during
+		// pulumi up. Perhaps a little forward-looking but it is nice to get this right.
 		if rplan.ChangeKind() == tfsandbox.ReplaceDestroyBeforeCreate {
 			resp.DeleteBeforeReplace = true
 		}
-		// TODO[pulumi/pulumi-terraform-module#100] populate replaces
-		resp.Replaces = []string{"todo"}
-	case tfsandbox.Create, tfsandbox.Read, tfsandbox.Delete, tfsandbox.Forget:
+
+		// The DiffResponse needs to be marked as a replace diff, either by having entries under DetailedDiff
+		// or having at least one entry under Replaces.
+		//
+		// TODO[pulumi/pulumi-terraform-module#100] detailed diffs is still left for later.
+		//
+		// For now just mark all changed keys as Pulumi understands them as Replaces.
+		changed, err := h.computeChangedKeys(req)
+		if err != nil {
+			return nil, err
+		}
+		if len(changed) > 0 {
+			resp.Replaces = propertyKeysAsStrings(changed)
+		} else {
+			// If Pulumi thinks there are no changed keys, this is still a replace and must be marked as
+			// such lest it renders as an update. Pulumi seems to tolerate non-existent properties here
+			// without displaying them.
+			resp.Replaces = []string{"__unknown"}
```

Comment:
Could use a different word here.. this has nothing to do with unknowns, just we do not know which property, or perhaps something else in the provider triggered the replace. It is more like the replace reason is unknown.. How do we render it?